### PR TITLE
Bootstraps netcore build

### PR DESCRIPTION
**Pull request by marek-safar**
This does not build but puts placeholders for syncing in place

### CODE REVIEW

File: mcs/build/profiles/core.make [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+BOOTSTRAP_PROFILE = build
+
+BOOTSTRAP_MCS = mcs
+MCS = $(BOOTSTRAP_MCS)
+
+PLATFORMS = macos linux win32 unix
+
+# nuttzing!
+
```
Comment (vargaz) [inline mcs/build/profiles/core.make]: Always wondered what this line means.

File: mcs/class/Makefile [hunk 2]
```diff
@@ -369,6 +369,9 @@ xbuild_4_0_dirs := \
 	Microsoft.Build.Tasks	\
 	Microsoft.Build
 
+core_dirs_parallell := \
+	NetCore
```
Comment (vargaz) [inline mcs/class/Makefile]: Typo.

File: configure.ac [hunk 3]
```diff
@@ -754,6 +754,13 @@ AC_ARG_WITH(mcs-path, [  --with-mcs-path=/path/to/mcs      Specify an alternate
 
 AC_ARG_WITH(jumptables, [  --with-jumptables=yes,no      enable/disable support for jumptables (ARM-only for now) (defaults to no)],[],[with_jumptables=no])
 
+AC_ARG_WITH(core,       [  --with-core=yes,no,only       controls whether to build Mono as a .NET Core runtime, not build it, or just build core (defaults to no)],[],[with_core=no])
```
[the remark this answers is not in the document]
Comment (migueldeicaza) [reply configure.ac]: Will add a README file that explains it in more detail.

I want the default to be just additive (ie, build the support in addition to everything I have).   The only will be the "Just build the bare minimum and remove the C support that is not needed", but I dont want to encourage that just yet

The reason for the latter is that I want to make it easy to reuse the same build across the current product and for core.   What i need to do is to fix the mcs/class Makefiles so we can do single-source builds for multiple builds (which we broke years ago, and I want to fix finally)

File: configure.ac [hunk 4]
```diff
@@ -754,6 +754,13 @@ AC_ARG_WITH(mcs-path, [  --with-mcs-path=/path/to/mcs      Specify an alternate
...
+if test x$with_core = xyes; then
```
[the remark this answers is not in the document]
Comment (migueldeicaza) [reply configure.ac]: Eventually, like I said :-)

**Comment by richlander** [comment]
Nice to see this project starting. Nice work.

**Pull request merged**